DigitalLearningSolutions.Web/Controllers/ForgotPasswordController.cs

```diff
@@ -1,21 +1,55 @@
 ﻿namespace DigitalLearningSolutions.Web.Controllers
 {
-    using Microsoft.AspNetCore.Mvc;
+    using DigitalLearningSolutions.Data.Exceptions;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Helpers;
     using DigitalLearningSolutions.Web.ViewModels.ForgotPassword;
-    using DigitalLearningSolutions.Web.ViewModels.LearningSolutions;
+    using Microsoft.AspNetCore.Mvc;
 
     public class ForgotPasswordController : Controller
     {
-        public IActionResult Index()
+        private readonly IPasswordResetService passwordResetService;
+
+        public ForgotPasswordController(IPasswordResetService passwordResetService)
         {
-            var model = new ForgotPasswordViewModel();
+            this.passwordResetService = passwordResetService;
+        }
+
+        [HttpGet]
+        public IActionResult Index(ForgotPasswordViewModel? model)
+        {
+            model ??= new ForgotPasswordViewModel();
             return View(model);
         }
 
         [HttpPost]
-        public IActionResult RecoverPassword(string emailAddress)
+        public IActionResult Index(string? emailAddress)
         {
-            // TODO: HEEDLS-354 - form submission logic
+            if (emailAddress?.Contains('@') != true)
+            {
+                ForgotPasswordViewModel model = GetEmailInvalidErrorModel(emailAddress ?? string.Empty);
+                return Index(model);
+            }
+
+            string baseUrl = ConfigHelper.GetAppConfig()["AppRootPath"];
+
+            try
+            {
+                passwordResetService.GenerateAndSendPasswordResetLink
+                (
+                    emailAddress,
+                    baseUrl
```

Comment:
Is there a reason we define the baseUrl here and pass it as a param, rather than define it when it needs to be used?

Reply:
As part of the previous set of corrections, the baseUrl is now retrieved from the appsettings under the web project. The data project does not have access to this. We could refactor the service so that the controller makes multiple calls and builds the link itself before passing that to the email sender?